tensorboard/webapp/runs/store/utils_test.ts

```diff
@@ -90,5 +90,150 @@ describe('run store utils test', () => {
+          [
+            buildRun({id: 'eid1/alpha', name: 'foo1bar1'}),
+            buildRun({id: 'eid1/beta', name: 'foo1bar2'}),
+            buildRun({id: 'eid2/beta', name: 'foo2bar1'}),
+            buildRun({id: 'eid2/gamma', name: 'gamma'}),
+          ],
+          {
+            'eid1/alpha': 'eid1',
+            'eid1/beta': 'eid1',
+            'eid2/beta': 'eid2',
+            'eid2/gamma': 'eid2',
+          }
+        );
+        expect(actual).toEqual({
+          matches: {},
+          nonMatches: [],
+        });
+      });
+
+      it('groups runs by regex without capture group', () => {
+        const actual = groupRuns(
+          {key: GroupByKey.REGEX, regexString: 'foo\\d+bar'},
+          [
+            buildRun({id: 'eid1/alpha', name: 'foo1bar1'}),
+            buildRun({id: 'eid1/beta', name: 'foo1bar2'}),
+            buildRun({id: 'eid2/beta', name: 'foo2bar1'}),
+            buildRun({id: 'eid2/gamma', name: 'gamma'}),
+          ],
+          {
+            'eid1/alpha': 'eid1',
+            'eid1/beta': 'eid1',
+            'eid2/beta': 'eid2',
+            'eid2/gamma': 'eid2',
+          }
+        );
+
+        expect(actual).toEqual({
+          matches: {
+            pseudo_group: [
+              buildRun({id: 'eid1/alpha', name: 'foo1bar1'}),
+              buildRun({id: 'eid1/beta', name: 'foo1bar2'}),
+              buildRun({id: 'eid2/beta', name: 'foo2bar1'}),
+            ],
+          },
+          nonMatches: [buildRun({id: 'eid2/gamma', name: 'gamma'})],
+        });
+      });
+
+      it('groups runs by regex with one capture group', () => {
+        const actual = groupRuns(
+          {key: GroupByKey.REGEX, regexString: 'foo(\\d+)bar'},
+          [
+            buildRun({id: 'eid1/alpha', name: 'foo1bar1'}),
+            buildRun({id: 'eid1/beta', name: 'foo1bar2'}),
+            buildRun({id: 'eid2/beta', name: 'foo2bar1'}),
+            buildRun({id: 'eid2/gamma', name: 'foo2bar3'}),
+            buildRun({id: 'eid2/alpha', name: 'alpha'}),
+          ],
+          {
+            'eid1/alpha': 'eid1',
+            'eid1/beta': 'eid1',
+            'eid2/beta': 'eid2',
+            'eid2/gamma': 'eid2',
+            'eid2/alpha': 'eid2',
+          }
+        );
+
+        expect(actual).toEqual({
+          matches: {
+            '["foo1bar","1"]': [
+              buildRun({id: 'eid1/alpha', name: 'foo1bar1'}),
+              buildRun({id: 'eid1/beta', name: 'foo1bar2'}),
+            ],
+            '["foo2bar","2"]': [
+              buildRun({id: 'eid2/beta', name: 'foo2bar1'}),
+              buildRun({id: 'eid2/gamma', name: 'foo2bar3'}),
+            ],
+          },
+          nonMatches: [buildRun({id: 'eid2/alpha', name: 'alpha'})],
+        });
+      });
+
+      it('groups runs by regex with multiple capture group', () => {
+        const actual = groupRuns(
+          {key: GroupByKey.REGEX, regexString: 'foo(\\d+)bar(\\d+)'},
```

Comment:
Could you modify the test by changing this regex string to `foo(\\d+)bar(\\d+).*`